### ISSUE TITLE
schutzbot: Come up with a clever way to check changes/PRs against the nightly test suite

### ISSUE THREAD

**Issue by atodorov** [osbuild/osbuild-composer#1326]
The nightly test suite relies on RPMs which have been prebuilt and executes via cron once per day. However there is a window of time between changes being merged via PR, RPMs landing into RHEL downstream and testing them via the nightly pipeline.  In rare cases we end up in a situation where the nightly pipeline will still fail eventhough PRs are all reporting green and everything we need has been backported. For example:

#1317 - that was needed because we had different places where source repositories are defined (which later override one another). That problem wasn't discovered until 28.1 was released and nightly tests were executed against 28.2. Even then it was needed to dig into the root cause to understand why we have the failure.

#1321 - is a direct descendant of 1317 (which was merged prematurely) and fixes a typo with an extra comma in the JSON file.

In short we don’t know if certain changes will continue working once they land into a nightly tree b/c we don’t test them in the same way. When it is suspected that this is necessary (e.g. I know that I'm modifying something related to the nightly pipeline) we have the mechanism to replay the test jobs started by PRs and instruct Schutzbot to think that it is running against the nightly tree. 

Opening this issue to not forget and investigate how to deal with this and whether or not we think it is necessary to change how we test in the first place.
